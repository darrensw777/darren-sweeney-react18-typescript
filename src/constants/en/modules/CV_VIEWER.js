export const CV_VIEWER = {
    DOWNLOAD: "Click here to download a PDF of my CV",
    CONTACT: "Contact me",
    PAGE: "Page",
    OF: "of",
    NEXT: "Next",
    PREVIOUS: "Previous",
};
