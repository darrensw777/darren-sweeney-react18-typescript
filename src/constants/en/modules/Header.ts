export const HeaderCopy = {
    headerLinks: [
        {
            linkTo: '/',
            label: 'Home',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
        {
            linkTo: '/about',
            label: 'About',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/work',
            label: 'Work',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/music',
            label: 'Music',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/videos',
            label: 'Videos',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/gallery',
            label: 'Gallery',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/contact',
            label: 'Contact',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
    ],
};

export default HeaderCopy;
