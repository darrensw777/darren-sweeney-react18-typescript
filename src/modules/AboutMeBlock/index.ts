/* istanbul ignore file */
export { default } from './AboutMeBlock';
