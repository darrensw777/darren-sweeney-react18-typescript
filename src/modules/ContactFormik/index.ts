/* istanbul ignore file */
export { default } from './ContactFormik';
