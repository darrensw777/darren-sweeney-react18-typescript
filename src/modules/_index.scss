@import './Header/Header.scss';
@import './Footer/Footer.scss';
@import './PageHeader/page-header.scss';
@import './AboutMeBlock/about-me-block.scss';
@import './CvViewer/cv-viewer.scss';
@import './SoundCloudPlayer/soundcloud-player.scss';
@import './ImagesGallery/images-gallery.scss';
@import './ContactFormik/contact-formik.scss';
@import './InterestLists/interest-lists.scss';