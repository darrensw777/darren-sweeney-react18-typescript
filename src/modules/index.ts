/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
