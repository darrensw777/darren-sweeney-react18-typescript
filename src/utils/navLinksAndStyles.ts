const navLinksAndStyles = [
    {
        link: "/",
        title: "Home",
        id: "Home",
        iconClass: "fa fa-home",
        backgroundColor: "#2C1320",
        homeImagePath: "",
        navBlock: false,
    },
    {
        link: "/about",
        title: "About me",
        id: "About",
        iconClass: "fa fa-info-circle",
        backgroundColor: "#1abc9c",
        homeImagePath: "/img/gallery/darren.webp",
        navBlock: true,
    },
    {
        link: "/work",
        title: "Work/CV",
        id: "Work",
        iconClass: "fa fa-code",
        backgroundColor: "#3498db",
        homeImagePath: "/img/gallery/zt4.webp",
        navBlock: true,
    },
    {
        link: "/music",
        title: "Music",
        id: "Music",
        iconClass: "fa fa-music",
        backgroundColor: "#9b58b5",
        homeImagePath: "/img/gallery/singing3.webp",
        navBlock: true,
    },
    {
        link: "/gallery",
        title: "Gallery",
        id: "Pictures",
        iconClass: "fa fa-picture-o",
        backgroundColor: "#e67e22",
        homeImagePath: "/img/gallery/zt7.webp",
        navBlock: true,
    },
    {
        link: "/Contact",
        title: "Contact",
        id: "Contact",
        iconClass: "fa fa-envelope-o",
        backgroundColor: "#d9a81d",
        homeImagePath: "/img/gallery/benidorm5.jpg",
        navBlock: false,
    },
];

export default navLinksAndStyles;
