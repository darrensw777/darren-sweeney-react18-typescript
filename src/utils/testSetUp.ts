import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
