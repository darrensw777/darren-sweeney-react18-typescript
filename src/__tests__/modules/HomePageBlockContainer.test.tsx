import { HomePageBlockContainer } from 'modules';
import testSetUp from 'utils/testSetUp';
import navLinksAndStyles from "utils/navLinksAndStyles";

describe('HomePageBlockContainer container', () => {
    const copyBlock = {
        general: ['hello', 'world'],
        greetings: {},
        interests: [
            {
                question: 'string',
                answers: ['answer1'],
            },
        ],
    };

    let asFragment: any, container: any;
    beforeEach(() => {
        ({ asFragment, container } = testSetUp({
            Component: <HomePageBlockContainer navLinksAndStyles={navLinksAndStyles} />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the block', () => {
        const blocks = container.getElementsByClassName('home-group-container');
        expect(blocks.length).toBe(1);
    });
});
