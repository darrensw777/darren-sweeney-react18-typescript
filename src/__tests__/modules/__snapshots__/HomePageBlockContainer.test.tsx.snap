// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageBlockContainer container matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="moduleStyles__HomePageBlockContainerWrapper-sc-187oehp-1 jqUcdL"
  >
    <div
      class="home-group-container"
    >
      <div
        class="componentStyles__HomePageBlockWrapper-sc-1wsblha-1 kqpejr"
      >
        <div
          class="homepage-block"
        >
          <a
            class="link"
            href="/about"
          >
            <div
              class="card-image"
            >
              <img
                alt=""
                class="img-responsive"
                src="/img/gallery/darren.webp"
              />
            </div>
            <div
              class="link-holder"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
              About me
            </div>
          </a>
        </div>
      </div>
      <div
        class="componentStyles__HomePageBlockWrapper-sc-1wsblha-1 kqpejr"
      >
        <div
          class="homepage-block"
        >
          <a
            class="link"
            href="/work"
          >
            <div
              class="card-image"
            >
              <img
                alt=""
                class="img-responsive"
                src="/img/gallery/zt4.webp"
              />
            </div>
            <div
              class="link-holder"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
                />
              </svg>
              Work/CV
            </div>
          </a>
        </div>
      </div>
      <div
        class="componentStyles__HomePageBlockWrapper-sc-1wsblha-1 kqpejr"
      >
        <div
          class="homepage-block"
        >
          <a
            class="link"
            href="/music"
          >
            <div
              class="card-image"
            >
              <img
                alt=""
                class="img-responsive"
                src="/img/gallery/singing3.webp"
              />
            </div>
            <div
              class="link-holder"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
                />
              </svg>
              Music
            </div>
          </a>
        </div>
      </div>
      <div
        class="componentStyles__HomePageBlockWrapper-sc-1wsblha-1 kqpejr"
      >
        <div
          class="homepage-block"
        >
          <a
            class="link"
            href="/gallery"
          >
            <div
              class="card-image"
            >
              <img
                alt=""
                class="img-responsive"
                src="/img/gallery/zt7.webp"
              />
            </div>
            <div
              class="link-holder"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
                />
              </svg>
              Gallery
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
